src/main/java/com/hazelcast/gcp/GcpConnectionException.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.hazelcast.gcp;
+
+import com.hazelcast.internal.json.Json;
+import com.hazelcast.internal.json.JsonObject;
+import com.hazelcast.internal.json.JsonArray;
+import com.hazelcast.internal.json.JsonValue;
+
+/**
+ * Exception to indicate an issue when pinging GCP API.
+ *
+ * A list of errors can be found at: {@see https://cloud.google.com/resource-manager/docs/core_errors}.
+ */
+
+public class GcpConnectionException extends RuntimeException {
```

Comment:
I wonder how much value is in this class and all the error parsing activities. Could you paste here comparison of a sample message: 
- your version
- if we just display the whole message from GCP

The thing I wonder about is if it's worth to parse this message at all. If it's comprehensive to the end user without parsing, then we could skip this at all.

Reply:
@leszko Thanks for the feedback! Here are the comparisons of the error messages:

With GcpConnectionException parsing for default service account API scope error message:
<img width="1671" alt="GCPExceptionLogging" src="https://user-images.githubusercontent.com/22038855/81029893-927b7300-8e3b-11ea-9ec7-6497a8010dd7.png">

Without GcpConnectionException parsing for default service account API scope error message:
<img width="1143" alt="GcpErrorOutput" src="https://user-images.githubusercontent.com/22038855/81029907-a1622580-8e3b-11ea-8f38-fde3b08fbd6c.png">

Without GcpConnectionException parsing for custom service account permissions error message:
<img width="1015" alt="GcpErrorServiceAccountOutput" src="https://user-images.githubusercontent.com/22038855/81029930-bb036d00-8e3b-11ea-8e06-130c6468636d.png">

The main value I see in the class is being able to parse the message so as not to retry the method call, and to give a more directed solution for how to resolve the error. As it stands, with the implementation of the retryUtils.retry() method checking for the nonRetryableKeywords and if I remove the conversion from the RestClientException to a GcpConnectionException, it is not able to parse the json error string for "Request had insufficient authentication scopes". As you can see in the screenshot, when the API access scope for the Compute Engine is not read-only, it will continue retrying the GcpClient.fetchGcpAddresses(). Let me know your thoughts - I'm happy to either remove the class or keep it in there.

Reply:
@ryanlindeborg Thanks a lot for the comparison. I get your point. However, tbh I think it's better to remove this class. I know that then we'll need to retry the exception. But still, I think it adds a lot of complexity and the exception is already clear enough.

So, sorry to ask you for that, but could you remove this and then I'll review this PR again and hopefully we'll get closer to merging it? :)